components/brave_wallet/browser/brave_wallet_utils_unittest.cc

```diff
@@ -44,6 +44,15 @@ TEST(BraveWalletUtilsUnitTest, PadHexEncodedParameter) {
   ASSERT_EQ(
       out,
       "0x0000000000000000000000004e02f254184E904300e0775E4b8eeCB14a1b29f0");
+
+  // Corner case: 62
+  ASSERT_TRUE(PadHexEncodedParameter(
+      "0x11111111112222222222333333333344444444445555555555666666666600",
```

Comment:
I was reviewing something else, but occasionally found two issues in this code. Low-level manipulations with bytes are very error-prone, so I also tried to simplify code and minimize juggling with indices/counts/sizes